### ISSUE TITLE
Error during NAMD build: "charmc: line 236: --gnu_extensions: command not found"

### ISSUE THREAD

**Issue by shlgf** [UIUC-PPL/charm#3749]
I'm encountering an error while building NAMD  from source using `./build charm++ ucx-linux-x86_64 ompipmix --with-production --force`. The error message I'm getting is:

```
Scanning dependencies of target ck
[ 22%] Building CXX object src/ck-core/CMakeFiles/ck.dir/__/ck-cp/API-generator/cp_effects.C.o
pgc++-Info-Switch -fast -Mvect -fast -Mvect -Mvect forces -O3
pgc++-Warning-No files to process

/mnt/PublicSoftware/NAMD/NAMD_3.0b3_Source/charm-v7.0.0/ucx-linux-x86_64-ompipmix/bin/charmc: line 236: --gnu_extensions: command not found
Fatal Error by charmc in directory /mnt/PublicSoftware/NAMD/NAMD_3.0b3_Source/charm-v7.0.0/ucx-linux-x86_64-ompipmix/src/ck-core
   Command pgc++ -fPIC -DCMK_FIND_FIRST_OF_PREDICATE=1 --no_using_std -DCMK_GFORTRAN -I/mnt/PublicSoftware/NAMD/NAMD_3.0b3_Source/charm-v7.0.0/ucx-linux-x86_6pipmix/include -I/usr/include/ -I./proc_management/ -I./proc_management/simple_pmi/ -D__CHARMC__=1 -DAMPI_ERRHANDLER_RETURN=0 -I/mnt/PublicSoftware/NAMD/NAMD_3_Source/charm-v7.0.0/src/ck-core/. -I/mnt/PublicSoftware/NAMD/NAMD_3.0b3_Source/charm-v7.0.0/src/ck-core/../ck-ldb -I/mnt/PublicSoftware/NAMD/NAMD_3.0b3_Sourharm-v7.0.0/src/ck-core/../ck-perf -I/mnt/PublicSoftware/NAMD/NAMD_3.0b3_Source/charm-v7.0.0/src/ck-core/../ck-cp -I/mnt/PublicSoftware/NAMD/NAMD_3.0b3_Sourcerm-v7.0.0/src/ck-core/../util/topomanager -I/mnt/PublicSoftware/NAMD/NAMD_3.0b3_Source/charm-v7.0.0/src/ck-core/../conv-perf -DNDEBUG -fast -O3 --c++11 --gnu_nsions -O2 -U_FORTIFY_SOURCE --c++11;--gnu_extensions -c /mnt/PublicSoftware/NAMD/NAMD_3.0b3_Source/charm-v7.0.0/src/ck-cp/API-generator/cp_effects.C -o CMakes/ck.dir/__/ck-cp/API-generator/cp_effects.C.o returned error code 127
charmc exiting...
make[2]: *** [src/ck-core/CMakeFiles/ck.dir/build.make:815: src/ck-core/CMakeFiles/ck.dir/__/ck-cp/API-generator/cp_effects.C.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1991: src/ck-core/CMakeFiles/ck.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```
 I noticed that in the `conv-mach-opt.sh` file located at `/ucx-linux-x86_64-ompipmix/include/`, there is consistent text that matches the problematic compiler command "--c++11;--gnu_extensions". It seems that an extra semicolon is present between "--c++11" and "--gnu_extensions".

I suspect this extra semicolon might be causing the issue. However, each time I build, this file is recreated, and I'm unsure where to make the necessary modification to address this problem.

Could you please advise on how to resolve this issue? Is there a specific file or configuration that I should modify to avoid this error? Any guidance or suggestions would be greatly appreciated.

**Comment by evan-charmworks** [comment]
I don't see the text `gnu_extensions` any where in the code history of Charm or NAMD, which is odd.

@matthiasdiener Could CMake be adding this to the arguments passed to charmc, because of these commands we run? 

```cmake
# We need C++11 for (almost) all targets
set(CMAKE_CXX_STANDARD 11)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
```

```cmake
# Workaround for CMake bug when calling check_include_file_cxx() with C++11:
# https://stackoverflow.com/uestions/47213356/cmake-using-corrext-c-standard-when-checking-for-header-files
set(CMAKE_REQUIRED_FLAGS "-std=c++11")
```

The [Summit User Guide](https://docs.olcf.ornl.gov/systems/summit_user_guide.html#c-compilations) recommends `-std=c++11 -gnu_extensions` for PGI. Could there be something wrong that causes a semicolon to appear there instead of a space? We have this code too:

```cmake
separate_arguments(MY_EXTRA_OPTS UNIX_COMMAND "${EXTRA_OPTS}")
add_compile_options("${MY_EXTRA_OPTS}")
string(REPLACE ";" " " MY_EXTRA_OPTS "${MY_EXTRA_OPTS}")
set(OPTS "${OPTS} ${MY_EXTRA_OPTS}")
set(OPTSATBUILDTIME "${OPTSATBUILDTIME} ${MY_EXTRA_OPTS}")
```

```cmake
get_target_property(MAIN_CFLAGS ck COMPILE_OPTIONS)
string(REPLACE ";" " " MAIN_CFLAGS "${MAIN_CFLAGS}")
```

**Comment by evan-charmworks** [comment]
@shlgf Does the same build error occur when you try building the latest main branch of Charm++?

**Comment by evan-charmworks** [comment]
I was able to reproduce this issue and fix it, at least in my testing, as part of #3757.
